### DISCUSSION TITLE
Connection to inverter AP lost periodically same time

### DISCUSSION THREAD

**Discussion by ollila** [wlcrs/huawei_solar#730] [Setting up the integration]
I'm using the recommended connection mentod to inverter AP with WiFi USB stick on my HA. I'm experiencing recurring losses of connection every 1-3 days where around same time (very early morning) connection to inverter is lost. From logs I see the following error:

This error originated from a custom integration.

Logger: custom_components.huawei_solar
Source: helpers/update_coordinator.py:321
integration: Huawei Solar (documentation, issues)
First occurred: June 12, 2024 at 03:58:16 (4 occurrences)
Last logged: 05:22:07

Timeout fetching BTXXXXXXXX_data_update_coordinator data

HA shows the it network interface options it would still be connected to inverter access point but all entities are unavailale. Connection resumes when I manually reconnect to AP and will be solid until the next disconnect in 1-3 days on the same time.

Has anyone else experienced similar issues? Or does someone has experience automating forced reconnection to AP from sensor unavailability trigger to recover from disconnect.

**Comment by ollila** [comment]
The mysterious disconnections ended after I had to recover from back up. The underlying reason was left as mystery since nothing should have changed when recovering from back up. In any case it's solved now and has been steable for a week.

**Comment by ManuelHecht** [comment]
I have absolutely the same problem. A disconnect every night and from there on, all Huawei entities are unavailable till I reconnect or reboot HA. All other integrations and entities are working fine. Maybe the connection is lost during my ISP reconnects my DSL and give me a new IP. But then the reconnect to the Huawei system should be done automatically, right.

What can I do? I'm new with HA. Sure I will try to restore the system from a backup, but I think this should not be the best practice.

Thanks

**Reply by LotharWoman** [reply]
I can confirm the issue with the loss after a DSL reconnect.
My provider disconnects every day at 4:00 AM and 40-50 minutes after that all Huawei entities are unviable for 30 minutes.
This has only been the case since I changed the router about a month ago. With my old router I kept seeing unaviable lasting a few seconds throughout the day.
With another router (TP-Link with OpenWRT) I get almost no connection via Modbus-TCP. Data only comes for a few minutes.
So I think it's a router problem and has nothing to do with the integration.

**Reply by ManuelHecht** [reply]
I don't think so. Because after a reboot of HA or a reconfiguration of the Huawei integration, all works fine again. I think it's a problem of the reconnection mechanism in the Huawei integration.

BTW: The reconnection worked till an HA or Huawei integration update two or three months ago. After this update, it starts to lose the connection every night. Before, it worked fine.

But I can't say witch update cases the problem.

**Reply by ollila** [reply]
> So I think it's a router problem and has nothing to do with the integration.

I had TP-Link Wi-Fi adapter connected directly to hardware I use to run HA without router involved. It can be wider HW issue but it’s not tied only to router as direct WiFi is also affected.

**Comment by ollila** [comment]
I took a closer look on the data and it seems the disconnection issue did not go away but automatic reconnection works within 1 minute. 

Also looking at the data there is link between disconnection time and inverter coming online. The disconnection time is not constant but it constantly happens few minutes before inverter starts to detect irradiation. As an example here are early and late disconnect times within lastt week:
![image](https://github.com/user-attachments/assets/7d318fec-6bfd-4572-a9bf-8c6d56bbef83)
![image](https://github.com/user-attachments/assets/c27736cf-6330-484b-9932-0021ac23edde)

Interestingly the disconnect happens always before inverter waking up although the time varies day to day. This suggest its not just an router/wifi-stick issue.

**Comment by LotharWoman** [comment]
Here it is like this. Every night after requesting a new IP from the provider, it takes a few minutes and the connection via Modbus is interrupted for 30 minutes.
Even with external modbus-software, no query is possible during these 30 minutes and will be acknowledged with a timeout.
![Unbenannt 1](https://github.com/user-attachments/assets/5f0eff78-ec9c-4460-bce0-a8b7ea62b5fc)


And here again from just now. I got a new external IP at 3:49 p.m.
![Unbenannt 2](https://github.com/user-attachments/assets/dcdc48f9-1eb9-4055-a6b9-247aae58a0db)
